utils/prepare_schema.js

```diff
@@ -54,63 +66,76 @@ const index_check = (values_array, object) => {
       })
       return true
     }
+
     return false
   })
+
   if (index) {
-    object.index = index
-    object.tokenizer = tokenizer_array
+    return {
+      ...object,
+      index,
+      tokenizer: tokenizer_array,
+    }
   }
-}
 
+  return object
+}
 const other_options = (values_array, object) => {
+  let new_object = object
   values_array.forEach(value => {
-    if (value.includes('@upsert'))
-      object.upsert = true
-    else if (value.includes('@lang'))
-      object.lang = true
+    if (value.includes('@upsert')) {
+      new_object = {
+        ...object,
+        upsert: true,
+      }
+    } else if (value.includes('@lang')) {
+      new_object = {
+        ...object,
+        lang: true,
+      }
+    }
   })
+  return new_object
 }
```

Comment:
same here, the goal is not to replace your old code with the same design, but to think differently :) **mutate less, compose more**